client/store.js

```diff
@@ -25,15 +26,28 @@ const setCategories = categories => ({
   categories,
 });
 
+const setReviews = reviews => ({
+  type: SET_REVIEWS,
+  reviews,
+});
+
 //THUNK CREATORS
-export const fetchProducts = () => async dispatch => {
-  try {
-    const response = await axios.get('api/products');
-    const products = response.data;
-    return dispatch(setProducts(products));
-  } catch (error) {
-    throw new Error(error);
-  }
+export const fetchProducts = () => {
+  return dispatch => {
+    return axios
+      .get('api/products')
+      .then(res => res.data)
+      .then(products => dispatch(setProducts(products)));
+  };
+};
+
+export const fetchProduct = id => {
+  return dispatch => {
+    return axios
+      .get(`api/products/${id}`)
+      .then(res => res.data)
+      .then(product => dispatch(setProducts([product])));
```

Comment:
I think best practice to use an action type and action creator for only one purpose and not try to reuse a plural products for singular product.  That could get confusing.

Create an action type and action creator for each type:
```
const SET_PRODUCT = 'SET_PRODUCT';

const setProduct = product => ({ type: SET_PRODUCT, product });
```
Also add a reducer, `product`.  We will also be using this reducer later for adding and deleting a product.
 

```suggestion
      .then(product => dispatch(setProducts(product)));
```